### PR TITLE
local-bd-etcd: fix CPMS

### PR DESCRIPTION
**Pull request by EmilienM**
* Re-use code from cpms.go
* Simplify error handling in cpms.go for errors out of getCPMS.
  It wasn't useful to count the number of CPMS, it'll always be zero
  when an error occurs.
* If an error is found when listing CPMS, just skip the test.
  We don't care how many CPMS are found, there is already a check
  in the function. This test can only function when one CPMS is found. OpenShift CPMS can only have 1 CPMS for now.
* Stop returning numCpms, we don't need to know at it should always
  be one, otherwise we return an error.
* Increase the size of the block device required for etcd. etcd grow to
  max 8GB, so let's have some window and require 10 minimum.

### CODE REVIEW

File: test/extended/openstack/local-bd-etcd.go
Comment (itzikb-redhat) [inline test/extended/openstack/local-bd-etcd.go:33]: I have getControlPlaneMAchineSet at cpms.go https://github.com/openshift/openstack-test/pull/124/commits/cb55d18920ac8cd2724122e427f17709d6d45f35#diff-7b12ea0dbd387e5b0d5a4eef6d705422cd7eabd7cd8093be463d17d9eb882841R132 , Please take a look and tell me if you think it should be removed/moved etc

Comment (EmilienM) [reply test/extended/openstack/local-bd-etcd.go:33]: excellent, I'll refactor :-)

File: test/extended/openstack/cpms.go
Comment (EmilienM) [inline test/extended/openstack/cpms.go:46]: this was not useful. in case of an error, `numCpms` will always be `0`. Let's just skip the test if CPMS can't be found (in case of UPI for example).

File: test/extended/openstack/cpms.go
Comment (MaysaMacedo) [inline test/extended/openstack/cpms.go:46]: If any error happen we would skip the test, is this really desired?

Comment (EmilienM) [reply test/extended/openstack/cpms.go:46]: yes, we rely on CPMS data, so let's skip the test if we can't get it.

Comment (MaysaMacedo) [reply test/extended/openstack/cpms.go:46]: disregard my comment. I didn't see your last comment before I commented

**Comment by MaysaMacedo** [comment]
/lgtm

let's see result of jobs

**Comment by MaysaMacedo** [comment]
/lgtm
wait for CI before approval

**Comment by MaysaMacedo** [comment]
/approve

All jobs are required anyway

**Comment by EmilienM** [comment]
@pierreprinetti @itzikb-redhat See the change in `cpms.go`; I think I simplified it a bit. Also note that I made the change to cpms.go into a separate commit so this can be cherry-picked without the other file, since the etcd tests is only for 4.15 and beyond.

@MaysaMacedo I think I've fixed the error we discussed offline, we'll see how CI works now.

**Comment by EmilienM** [comment]
I tested locally with a deployment with etcd on additional disk and the test was skipped so something is wrong...

I'll look into it tomorrow. Sorry for the UPI job breakage in the meantime.

**Comment by EmilienM** [comment]
```
passed: (6.5s) 2023-10-31T01:10:19 "[sig-installer][Suite:openshift/openstack] The OpenShift cluster runs with etcd on ephemeral local block device"
```
Got it working :-)

File: test/extended/openstack/cpms.go
Comment (itzikb-redhat) [reply test/extended/openstack/cpms.go:46]: Is it possible that due to a bug there will be more than one CPMS?

File: test/extended/openstack/local-bd-etcd.go
Comment (itzikb-redhat) [inline test/extended/openstack/local-bd-etcd.go:94]: You may want to declare  the provider spec value spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value once

Comment (EmilienM) [reply test/extended/openstack/local-bd-etcd.go:94]: done

**Comment by EmilienM** [comment]
Tested with latest patchset:
```
passed: (5.9s) 2023-10-31T13:02:22 "[sig-installer][Suite:openshift/openstack] The OpenShift cluster runs with etcd on ephemeral local block device"
```

**Comment by EmilienM** [comment]
/hold cancel
I think this is good to go, we can always iterate later.
@itzikb-redhat I'll let you cherry-pick the second commit to 4.14 if you want.

**Review by MaysaMacedo — APPROVED** [review]

**Pull request merged**
